File: gaia-sdk-javascript/src/graphql/request/type/BehaviourEdge.ts

import {Behaviour} from "./Behaviour";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";

/**
 * Reresents a paginated behaviour
 */
export class BehaviourEdge extends Array<(_:VariableRegistry) => string> {

    public node = (config: (_:Behaviour) => void) => this.push((registry) => {
        const entity = new Behaviour();
        config(entity);
        return "node { " + entity.render(registry) + " }";
    });

    public cursor = () => { 
        this.push(_ => "cursor")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/BehaviourEdge.py

from gaia_sdk.graphql.request.type.Behaviour import Behaviour

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry


class BehaviourEdge(list):
    """
    Reresents a paginated behaviour
    """

    def node(self, config: Callable[['Behaviour'], None]):
        def callback(registry: VariableRegistry):
            entity = Behaviour()
            config(entity)
            return "node {" + entity.render(registry) + "}"
        self.append(callback)

    def cursor(self):
        self.append(lambda x: "cursor")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
